### DISCUSSION TITLE
How to nuxt preview and development as serverless function and production as static renderd

### DISCUSSION THREAD

**Discussion by falko-apheris** [vercel/community#1210] [Help]
Hello there my setup is Nuxt with DatoCMS. Since I am using DatoCMS as external data source I want to deploy my site differently for each of the following:

- production: static code with `nuxt generate` (for the speed and SEO benefits)
- development (PR): serve as serverless function using CMS preview data (so I can test the branch with different CMS values) 
- preview: serve as serverless function using CMS preview data (same as above)

I have managed to run a preview domain with a serverless function by using 
vercel.json
```
{
"builds": [
		{
			"src": "nuxt.config.js",
			"use": "@nuxtjs/vercel-builder",
			"config": {}
		}
	]
}
```
and a different `robots.txt` file in a special branch called `preview`. The CMS datasource is switched to preview based on a enviornment flag.

The production deployment uses the `main` branch and renders a static site with `nuxt generate`.

I wonder if I could switch the building process depending on env variables specific for production, dev, and preview deployments but not use separate branches for that (since this is not really possible for the development deployments generated for each pull request, as I usually branch of from the `main` branch).

I hope this makes sense, I would appreciate any help.

**Comment by swarnava** [comment]
Hi there,

You can use [Ignore Build step](https://vercel.com/support/articles/how-do-i-use-the-ignored-build-step-field-on-vercel) to target build command based on different environment or branch. Here is a great article to get started: https://www.storyblok.com/tp/different-build-commands-branch-name-vercel

**Comment by amyegan** [comment]
If you haven't already, I think it's also worth asking on [Nuxt Discord](https://discord.com/invite/ps2h6QT) or in the [Nuxt framework repo](https://github.com/nuxt/framework/discussions) for advice on this. Someone there might already be doing something similar and be able to share their experience!

**Comment by falko-apheris** [comment]
Thank you for the answers so far but from the linked documentation I do not understand how I can use a serverless function based on e.g. an environment variable. I do not see an option how to set up a vercel runtime in via a script.

What I would need is to set `buildCommand`  or `functions` depending on the enviornment.

vercel.json 
```
{
	"buildCommand": "nuxt generate --fail-on-error",
	"functions": {
		"nuxt.config.js": {
			"runtime": "@nuxtjs/vercel-builder"
		}
	}
}
```

I have also started to contact the nuxt community but did not get an answer so far.

**Reply by falko-apheris** [reply]
Another solution I could think of is to generate the `vercel.json` file on install. Could this work?
